### PR TITLE
LV-Update-Product

### LINKED ISSUE #10: [Core][Data][App] Update Product
Product Brand should not be a string, but an actual Brand object.
  
One Product can have one Brand 
And One Brand can have many Products

We are going to keep the existing xml file to build data 

If data generation needs to be performed in a certain order you can take advantage of a design pattern.

### PR DESCRIPTION
**Pull request by LVarsaminov**
# Summary
[Basic information about the purpose of the PR]
The product class now has an updated Brand property that is of type Brand instead of string. One brand may have many products but a product has only one brand. 

# Technical details
[any technical changes or details that need highlighting]
Added the brand class, changed the brand property to brand object, separated logic for extracting data and seeding the db into 2 services, have implemented the CoR design pattern using a specific order of execution and created unit tests for the services and the CoR correct execution.
# Test
[how should this PR be tested and things to watch out for]
Once you build the project you can check in the database if the brands and products tables have been updated. If they have been updated the base functionality works fine. Then run the unit tests for BrandService, ProductService and DataImportHander and they should all pass

### CODE REVIEW

File: SportsGoods.App/Helper/Orchestrator.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+﻿using SportsGoods.App.Services;
```
Comment (clangelov) [inline SportsGoods.App/Helper/Orchestrator.cs]: Delete this file, as no longer is needed
@LVarsaminov

File: SportsGoods.Data/SportsGoods.Data.csproj [hunk 2]
```diff
@@ -10,6 +10,7 @@
     <PackageReference Include="Microsoft.AspNetCore.Identity.EntityFrameworkCore" Version="8.0.3" />
     <PackageReference Include="Microsoft.EntityFrameworkCore" Version="8.0.3" />
     <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="8.0.3" />
+    <PackageReference Include="Newtonsoft.Json" Version="13.0.1" />
```
Comment (clangelov) [inline SportsGoods.Data/SportsGoods.Data.csproj]: The package is not used 
Remove it 
@LVarsaminov

File: SportsGoods.App/Services/BrandService.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+using SportsGoods.Data.DAL;
+using System.Xml.Linq;
+
+namespace SportsGoods.App.Services
+{
+    public class BrandService : IBrandService
+
+    {
+        private readonly ApplicationDbContext _context;
+        private readonly IBrandRepository _brandRepository;
```
Comment (clangelov) [inline SportsGoods.App/Services/BrandService.cs]: This one is not used in the class 
remove it 
@LVarsaminov

File: UnitTests/Tests/BrandServiceTest.cs [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    [TestFixture]
+    public class BrandServiceTest
+    {
+        private static ApplicationDbContext _context = null!;
+        private ApplicationDbContext _testContext = null!;
+
+        [OneTimeSetUp]
+        public static async Task OneTimeSetUp()
+        {
+            var connectionString = "Server=.;Database=SportsGoods;Trusted_Connection=True;TrustServerCertificate=True;";
```
Comment (clangelov) [inline UnitTests/Tests/BrandServiceTest.cs]: do not use the db of the application in tests 
@LVarsaminov

File: UnitTests/Tests/ProductServiceTest.cs [hunk 5]
```diff
@@ -12,21 +12,40 @@ namespace SportsGoods.App.Tests
...
 
-        [SetUp]
-        public async Task Setup()
-        {
-            var connectionString = "Server=(localdb)\\MSSQLLocalDB;Database=SportsGoodsTest;Trusted_Connection=True;TrustServerCertificate=True;";
 
+        [OneTimeSetUp]
+        public static async Task OneTimeSetUp()
+        {
+            var connectionString = "Server=.;Database=SportsGoods;Trusted_Connection=True;TrustServerCertificate=True;";
```
Comment (clangelov) [inline UnitTests/Tests/ProductServiceTest.cs]: Same thing here 
do not mix the tests with the actual app 
@LVarsaminov

**Review by clangelov — CHANGES_REQUESTED** [review]

File: SportsGoods.App/Helper/Orchestrator.cs (hunk 1 above)
Comment (LVarsaminov) [reply SportsGoods.App/Helper/Orchestrator.cs]: Done

File: SportsGoods.Data/SportsGoods.Data.csproj (hunk 2 above)
Comment (LVarsaminov) [reply SportsGoods.Data/SportsGoods.Data.csproj]: Done

File: UnitTests/Tests/BrandServiceTest.cs (hunk 4 above)
Comment (LVarsaminov) [reply UnitTests/Tests/BrandServiceTest.cs]: Done

File: UnitTests/Tests/ProductServiceTest.cs (hunk 5 above)
Comment (LVarsaminov) [reply UnitTests/Tests/ProductServiceTest.cs]: Done

**Review by clangelov — APPROVED** [review]
looks ok

**Pull request merged**